frontend/app/controllers/TierController.scala

```diff
@@ -43,19 +42,30 @@ trait UpgradeTier {
 
   def upgrade(tier: Tier) = MemberAction.async { implicit request =>
 
-    def futureCustomerOpt = request.member match {
+    def futurePaidPreviewOpt = request.member match {
       case paidMember: PaidMember =>
-        request.touchpointBackend.stripeService.Customer.read(paidMember.stripeCustomerId).map(Some(_))
+        for {
+          preview <- MemberService.previewUpgradeSubscription(paidMember, request.user, tier)
+          customer <- request.touchpointBackend.stripeService.Customer.read(paidMember.stripeCustomerId)
```

Comment:
@jennysivapalan exciting (or sad) fact about Scala future for-comprehensions is these two futures - `previewUpgradeSubscription` & `Customer.read` - will not run concurrently as written here. See the 'For comprehensions' section in this document http://danielwestheide.com/blog/2013/01/09/the-neophytes-guide-to-scala-part-8-welcome-to-the-future.html